WcaOnRails/app/models/competition.rb

```diff
@@ -130,6 +136,20 @@ class Competition < ApplicationRecord
 
   validates :currency_code, inclusion: { in: Money::Currency, message: proc { I18n.t('competitions.errors.invalid_currency_code') } }
 
+  validates_numericality_of :refund_policy_percent, greater_than_or_equal_to: 0, less_than_or_equal_to: 100, if: :automatic_registration_requirements?
+  validates :refund_policy_limit_date, presence: true, if: :refund_policy_percent?
+  validates_inclusion_of :on_the_spot_registration, in: [true, false], if: :automatic_registration_requirements?
+  validates_numericality_of :on_the_spot_entry_fee_lowest_denomination, greater_than_or_equal_to: 0, if: :on_the_spot_registration? && :automatic_registration_requirements?
+  monetize :on_the_spot_entry_fee_lowest_denomination,
+           as: "on_the_spot_base_entry_fee",
+           allow_nil: true,
+           with_model_currency: :currency_code
+  validates_numericality_of :guests_entry_fee, greater_than_or_equal_to: 0, if: :automatic_registration_requirements?
+  monetize :guests_entry_fee,
+           as: "guests_base_fee",
```

Comment:
I don't think we need to call this `guests_base_fee`. The idea of a "base" fee is for competitors, as they might have to pay more for individual events. I don't think that applies to guests.

Reply:
Guests could have to pay a fee per day or something similar, that's why I went with the "base" here.